Client/Frontend/Browser/MainMenuActionHelper.swift

```diff
@@ -356,34 +358,30 @@ class MainMenuActionHelper: PhotonActionSheetProtocol, FeatureFlaggable, CanRemo
                                      iconString: ImageIdentifiers.sendToDevice) { _ in
-            guard let bvc = self.menuActionDelegate as? InstructionsViewDelegate & DevicePickerViewControllerDelegate
+            guard let bvc = self.menuActionDelegate as? sendDelegate,
+                  let selectedTab = self.selectedTab,
+                  let url = selectedTab.url
             else { return }
 
-            if !self.profile.hasAccount() {
-                let colors = self.themeManager.currentTheme.colors
-                let instructionsView = InstructionsView(backgroundColor: colors.layer1,
-                                                        textColor: colors.textPrimary,
-                                                        imageColor: colors.iconPrimary,
-                                                        dismissAction: {
-                    bvc.dismissInstructionsView()
-                })
-                let hostingViewController = UIHostingController(rootView: instructionsView)
-                let navigationController = UINavigationController(rootViewController: hostingViewController)
-                navigationController.modalPresentationStyle = .formSheet
-                self.delegate?.showViewController(viewController: navigationController)
-                return
-            }
+            let themeColors = self.themeManager.currentTheme.colors
+            let colors = SendToDeviceHelper.Colors(defaultBackground: themeColors.layer1,
+                                                   textColor: themeColors.textPrimary,
+                                                   iconColor: themeColors.iconPrimary)
+
+            let shareItem = ShareItem(url: url.absoluteString,
+                                      title: selectedTab.title,
+                                      favicon: selectedTab.displayFavicon)
+            let helper = SendToDeviceHelper(shareItem: shareItem,
+                                            profile: self.profile,
+                                            colors: colors,
+                                            delegate: bvc)
+            let viewController = helper.initialViewController()
 
-            let devicePickerViewController = DevicePickerViewController()
-            devicePickerViewController.pickerDelegate = bvc
-            devicePickerViewController.profile = self.profile
-            devicePickerViewController.profileNeedsShutdown = false
-            let navigationController = UINavigationController(rootViewController: devicePickerViewController)
-            navigationController.modalPresentationStyle = .formSheet
             TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .sendToDevice)
-            self.delegate?.showViewController(viewController: navigationController)
+            self.delegate?.showViewController(viewController: viewController)
```

Comment:
nit, I don't think we need to create a `let viewController = helper.initialViewController()`. There's some other occurences of that in other methods
```suggestion
            self.delegate?.showViewController(viewController: helper.initialViewController())
```